src/services/WalletService/wallets/ledger/Ledger.ts

```diff
@@ -50,4 +51,38 @@ export default abstract class Ledger extends HardwareWallet {
 
     return response.address;
   }
+
+  public async getMultipleAddresses(dpaths: ExtendedDPath[]): Promise<DWAccountDisplay[]> {
+    if (dpaths.length === 0) {
+      console.error('[getMultipleAddresses]: Derivation paths not found');
+      return [];
+    }
+    return this.getDPathAddresses(dpaths);
+  }
+
+  public async getDPathAddresses(dpaths: ExtendedDPath[]): Promise<DWAccountDisplay[]> {
```

Comment:
1. Is `getDpathAddresses` used elsewhere to justify it being separate from `getMultipleAddresses` ?

2. Test with `spyOn`
```
 let person = new Person('Lorem', 'Ipsum');
 let spy = jest.spyOn(person, 'sayMyName').mockImplementation(() => 'Hello');
```

3. Once tested, we can most likely optimise with `Promise.all`